stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/images/menu/ImagesMenuAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package io.getstream.chat.android.ui.images.menu
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import coil.load
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamUiItemAttachmentImageBinding
+
+internal class ImagesMenuAdapter(
+    private val imageList: List<String>,
+    private val clickListener: (Int) -> Unit
+) : ListAdapter<String, ImagesMenuAdapter.ImageViewHolder>(diffUtil()) {
```

Comment:
We should use `submitList` to provide the items to `ListAdapter`, if we want its diffing functionality, because it's supposed to maintain its own list internally, and we have our own here. Or we can just use a regular `RecyclerView.Adapter` if we don't need the diffing at all in this case.